PixelHunter1995/GameStates/Exploring.cs

```diff
@@ -58,5 +60,20 @@ private void HandleDogs(InputManager input)
                 GameManager.Instance.StartDialog();
             }
         }
+
+        private void HandlePortals(InputManager input)
+        {
+            Portal portal;
+            if (!Scene.GetPortalAtCursor(input, out portal))
```

Comment:
Is this how you're supposed to do things? My natural instinct would be to just return a null value, and check if it's null here. Perhaps that is more Pythonic rather than C#-ic.